modules/activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/ThrowSignalEventSessionManager.java

```diff
@@ -0,0 +1,54 @@
+package org.activiti.engine.impl.persistence.entity;
+
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.function.Function;
+
+import org.activiti.engine.impl.persistence.AbstractManager;
+import org.activiti.engine.impl.pvm.delegate.ActivityExecution;
+
+/**
+ * Keep track of fired events for process instance execution 
+ * in the current transaction context 
+ *
+ */
+public class ThrowSignalEventSessionManager extends AbstractManager {

```

Comment:
Is it necessary to implement AbstractManager.class?
It does not seem to be persisting the state of the signal in the database.
AbstractManager.class is the abstract class for managing the entity (although there are some exceptions such as LDAPGroupManager.class and LDAPUserManager.class).

If you just want to keep the state of the signal in CommandContext, will you be able to use the attributes of CommandContext?
https://github.com/Activiti/Activiti/blob/5.x/modules/activiti-engine/src/main/java/org/activiti/engine/impl/interceptor/CommandContext.java#L83
https://github.com/Activiti/Activiti/blob/5.x/modules/activiti-engine/src/main/java/org/activiti/engine/impl/interceptor/CommandContext.java#L238
https://github.com/Activiti/Activiti/blob/5.x/modules/activiti-engine/src/main/java/org/activiti/engine/impl/interceptor/CommandContext.java#L245

Reply:
It is actually a good suggestion. I have removed ThrowSignalEventSessionManager in favor of CommandContext attributes. It makes the fix less invasive. :)